Tasks/KubeloginInstallerV0/utils.ts

```diff
@@ -0,0 +1,187 @@
+}
+
+export function resolvePlatform(): Platform {
+  const platform: string = process.platform;
+  const arch: string = process.arch;
+  if (platform === 'darwin' && arch === 'x64') {
+    return 'darwin-amd64';
+  } else if (platform === 'darwin' && arch === 'arm64') {
+    return 'darwin-arm64';
+  } else if (platform === 'linux' && arch === 'x64') {
+    return 'linux-amd64';
+  } else if (platform === 'linux' && arch === 'arm64') {
+    return 'linux-arm64';
+  } else if (platform === 'win32' && arch === 'x64') {
+    return 'win-amd64';
+  } else {
+    throw new Error(`Unsupported platform: ${platform}-${arch}`);
+  }
+}
+
+export function isLatestVersion(version: string): boolean {
+  const v: string = version.toLowerCase();
+  return v === 'latest' || v === '*' || v === '';
+}
+
+export async function getLatestVersionTag(): Promise<string> {
+  let request = new webClient.WebRequest();
+  request.uri = 'https://api.github.com/repos/' + KUBELOGIN_REPO_OWNER + '/' + KUBELOGIN_REPO + '/releases/latest';
+  request.method = 'GET';
+  request.headers = request.headers || {};
+  request.headers['User-Agent'] = userAgent;
+
+  const response = await webClient.sendRequest(request);
+  return response.body['tag_name'];
+}
+
+export async function getKubeloginRelease(version: string = 'latest', platform?: Platform): Promise<KubeloginRelease> {
+  const origVersion: string = version;
+  if (isLatestVersion(version)) {
+    version = await getLatestVersionTag();
+  }
+
+  version = toolLib.cleanVersion(version);
+  if (!version) {
+    throw new Error(taskLib.loc('Err_NotAValidSemverVersion'));
+  }
+
+  if (version[0] != 'v') {
+    version = 'v' + version;
+  }
+
+  platform = platform || resolvePlatform();
+
+  const releaseName: string = `kubelogin-${platform}.zip`;
+  const sha256: string = `${releaseName}.sha256`;
+
+  try {
+    let request = new webClient.WebRequest();
+    request.uri = 'https://api.github.com/repos/' + KUBELOGIN_REPO_OWNER + '/' + KUBELOGIN_REPO + '/releases/tags/' + version;
+    request.method = 'GET';
+    request.headers = request.headers || {};
+    request.headers['User-Agent'] = userAgent;
+
+    const response = await webClient.sendRequest(request);
+
+    const releaseUrl: string =
+      response.body['assets'].find(asset => {
+        return asset.name.includes(releaseName);
+      })?.browser_download_url || '';
+    const sha256Url: string =
+      response.body['assets'].find(asset => {
+        return asset.name.includes(sha256);
+      })?.browser_download_url || '';
+
+    return {
+      version,
+      platform,
+      name: releaseName,
+      releaseUrl: releaseUrl,
+      checksumUrl: sha256Url
+    };
+  } catch (err) {
+    taskLib.debug(err);
+    throw Error(taskLib.loc('Err_VersionNotFound', origVersion));
+  }
+}
+
+function getKubeloginDownloadPath(): string {
+  const tempDir: string = getTempDirectory();
+  const downloadPath: string = path.join(tempDir, KUBELOGIN_REPO + Date.now());
```

Comment:
Actually, I can remove it since I switched to downloadTool from toolLib. It adds guid to a path by default to ensure reliability. I used ArtifactEngine before.